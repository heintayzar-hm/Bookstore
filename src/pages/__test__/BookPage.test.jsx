import { describe, test } from 'vitest';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import BooksPage from '../BookPage';
import store from '../../state/store';
describe('BookPage', () => {
  test('renders the initial state correctly', () => {
    // Arrange

    // Act
    const { container } = render(
      <Provider store={store}>
        <BooksPage />
      </Provider>
      ,
    );
    // Assert
    expect(container).toMatchSnapshot();
  });
});
