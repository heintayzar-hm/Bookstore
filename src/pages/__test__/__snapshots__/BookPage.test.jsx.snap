// Vitest Snapshot v1

exports[`BookPage > renders the initial state correctly 1`] = `
<div>
  <ul>
    <div>
      <article>
        <span>
          1
        </span>
        <h2>
          How to eat
        </h2>
        <h5>
          Hein Tay Zar
        </h5>
        <button
          type="button"
        >
          Remove
        </button>
      </article>
    </div>
    <div>
      <article>
        <span>
          2
        </span>
        <h2>
          How to sleep
        </h2>
        <h5>
          Hein Tay Zar2
        </h5>
        <button
          type="button"
        >
          Remove
        </button>
      </article>
    </div>
  </ul>
  <article
    data-testid="form-container"
  >
    <h2>
      Add New Book
    </h2>
    <form>
      <input
        data-testid="book-title-input"
        placeholder="Book Title"
        required=""
        type="text"
        value=""
      />
      <input
        data-testid="author-title-input"
        placeholder="Author"
        required=""
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Book
      </button>
    </form>
  </article>
</div>
`;
